pkg/sql/temporary_schema_test.go

```diff
@@ -272,3 +329,47 @@ func TestTemporarySchemaDropDatabase(t *testing.T) {
 		assert.Equal(t, 0, tempObjectCount)
 	}
 }
+
+// ensureTemporaryObjectsAreDeleted ensures all the tempNames have been deleted.
+// This can take a longer amount of time if the job takes time.
+func ensureTemporaryObjectsAreDeleted(
+	ctx context.Context, t *testing.T, conn *sql.Conn, schemaName string, tempNames []string,
+) {
+	lease.TestingDisableTableLeases()
```

Comment:
put this in the test, not the helper -- you have to `defer lease.TestingDisableTableLeases()()` so that it is reset after the test.